### PR TITLE
138: adapt to new `mmrm` package version

### LINKED ISSUE #138: Replace use of `free_cores()`
To do:

- [x] In https://github.com/insightsengineering/tern.mmrm/blob/ce9ac5ee7949eb88043f78c96fa722cb5378a830/R/fit_mmrm.R#L153 replace `mmrm::free_cores()` with `parallelly::availableCores(omit = 1L)` as per https://github.com/openpharma/mmrm/blob/755222fc63c453ac42fab23d15c21e98cf9dd29f/NEWS.md#L22
- [x] Add package `parallelly` to Imports
- [x] Import a single function from `parallelly` to make `R CMD check` happy

### PR DESCRIPTION
**Pull request by danielinteractive**
closes #138

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -29,7 +29,8 @@ Imports:
     ggplot2,
     lifecycle,
     magrittr,
-    mmrm (>= 0.1.0),
+    mmrm (>= 0.2.2),
```
Comment (pawelru) [inline DESCRIPTION:32]: just to confirm
is it `>= 0.2.2` (i.e. last released version is fine) or rather `> 0.2.2` (i.e. development version)?

Comment (danielinteractive) [reply DESCRIPTION:32]: thx @pawelru, last released version is fine

File: tests/testthat/test-g_mmrm.R [hunk 2]
```diff
@@ -24,17 +24,17 @@ fit_mmrm_no_arms_object <- fit_mmrm(
 # g_mmrm_diagnostic ----
 
 test_that("g_mmrm_diagnostic works well with defaults", {
-  result <- expect_silent(g_mmrm_diagnostic(fit_mmrm_object))
+  result <- g_mmrm_diagnostic(fit_mmrm_object)
```
Comment (Melkiades) [inline tests/testthat/test-g_mmrm.R:27]: why expect_silent is no more needed in this test file? Do modifications produce warnings or plots now? We might want to keep those checks

File: tests/testthat/test-fit_mmrm.R [hunk 3]
```diff
@@ -37,6 +37,45 @@ test_that("fit_mmrm works with parallelization", {
...
+    vars = list(
+      response = "FEV1",
+      covariates = c("RACE", "SEX"),
+      id = "USUBJID",
+      arm = "ARMCD",
+      visit = "AVISIT"
+    ),
+    data = mmrm_test_data,
+    cor_struct = "unstructured",
+    optimizer = c("nlminb", "CG")
```
Comment (Melkiades) [inline tests/testthat/test-fit_mmrm.R:53]: optimzer extra arg (...) is no more documented

File: tests/testthat/test-fit_mmrm.R [hunk 4]
```diff
@@ -37,6 +37,45 @@ test_that("fit_mmrm works with parallelization", {
...
+    vars = list(
+      response = "FEV1",
+      covariates = c("RACE", "SEX"),
+      id = "USUBJID",
+      arm = "ARMCD",
+      visit = "AVISIT"
+    ),
+    data = mmrm_test_data,
+    cor_struct = "unstructured",
+    method = "Kenward-Roger"
```
Comment (Melkiades) [inline tests/testthat/test-fit_mmrm.R:72]: methodextra arg (...) is no more documented

File: R/fit_mmrm.R [hunk 5]
```diff
@@ -79,10 +76,6 @@ h_get_diagnostics <- function(fit) {
 #'   - `heterogeneous compound symmetry`: Heterogeneous Compound Symmetry covariance matrix, which uses
 #'        `T + 1` variance parameters.
 #'
-#'   For the `optimizer`, the user can choose among alternatives to the recommended `"automatic"`,
-#'   please see [mmrm::refit_multiple_optimizers()] for details. Usually it should not be necessary
-#'   to use a specific optimizer and the `"automatic"` setting should be kept.
-#'
```
Comment (Melkiades) [inline R/fit_mmrm.R:85]: is this information coming from `mmrm::mmrm()` documentation? if yes I think it is fine to reference it as it is done above (`@param ... additional arguments for [etc..`)

Comment (Melkiades) [reply R/fit_mmrm.R:85]: Otherwise, it could be added back into `@details` as it is done for `method`

File: tests/testthat/test-fit_mmrm.R (hunk 3 above)
Comment (Melkiades) [reply tests/testthat/test-fit_mmrm.R:53]: disregard this if this parameter is well documented in `mmrm::mmrm()`

File: tests/testthat/test-fit_mmrm.R (hunk 4 above)
Comment (Melkiades) [reply tests/testthat/test-fit_mmrm.R:72]: disregard this if this parameter is well documented in `mmrm::mmrm()`

**Review by Melkiades — CHANGES_REQUESTED** [review]
Everything looks fine to me, I have just a couple of doubts, then we can merge! :)

File: tests/testthat/test-g_mmrm.R (hunk 2 above)
Comment (danielinteractive) [reply tests/testthat/test-g_mmrm.R:27]: I forgot honestly. basically a message would be ok here, and a warning would anyway cause the test to fail

File: tests/testthat/test-fit_mmrm.R (hunk 3 above)
Comment (danielinteractive) [reply tests/testthat/test-fit_mmrm.R:53]: yes exactly, well documented in mmrm()

File: tests/testthat/test-fit_mmrm.R (hunk 4 above)
Comment (danielinteractive) [reply tests/testthat/test-fit_mmrm.R:72]: yes, well documented there

**Comment by danielinteractive** [comment]
thanks @Melkiades good questions, all ok I think

**Review by Melkiades — APPROVED** [review]
Lgtm! We can merge it now, thank you for the clarifications ;)

**Pull request merged**